wrappers/services.go

```diff
@@ -471,13 +471,48 @@ func tryFileUpdate(path string, desiredContent []byte) (old *osutil.MemoryFileSt
 	}
 }
 
-type SnapServiceOptions struct {
+type ServiceQuotaMap map[*snap.AppInfo]*quota.Group
+
+// MakeServiceQuotaMap builds a map of services and their quota groups. The goal is
+// to make sure that if any quota group applies to a snap service, then we can easily
+// look it up in this based on it's snap.AppInfo. We take the snap instance name, the services
+// want to include in the map (should only be services of the snap name provided) and the quota
+// group of the snap.
+func MakeServiceQuotaMap(snapName string, svcs []*snap.AppInfo, grp *quota.Group) ServiceQuotaMap {
+	if len(svcs) == 0 {
+		return nil
+	}
+
+	svcQuotaMap := make(ServiceQuotaMap, len(svcs))
+	for _, svc := range svcs {
+		// set nil grps for all services if parent is nil
+		if grp == nil {
+			svcQuotaMap[svc] = nil
+			continue
+		}
+
+		// always default to the snap quota group if the service is not
+		// in a separate one
+		svcGrp := grp.GroupForService(fmt.Sprintf("%s.%s", snapName, svc.Name))
+		if svcGrp == nil {
+			svcGrp = grp
+		}
+		svcQuotaMap[svc] = svcGrp
+	}
+	return svcQuotaMap
+}
+
+type SnapServicesOptions struct {
 	// VitalityRank is the rank of all services in the specified snap used by
 	// the OOM killer when OOM conditions are reached.
 	VitalityRank int
 
-	// QuotaGroup is the quota group for all services in the specified snap.
+	// QuotaGroup is the quota group for the specified snap.
 	QuotaGroup *quota.Group
+
+	// Services is the map of snap services that have been affected and should
+	// have their services updated.
+	Services ServiceQuotaMap
```

Comment:
so this should become ServiceQuotaGroup and be ok if it's only partially filled. It's also a more robust definition for an external API anyway.